plugin/json-message-codec.go

```diff
@@ -0,0 +1,20 @@
+package plugin
+
+import (
+	"encoding/json"
+)
+
+// JSONMessageCodec implements a MessageCodec using JSON for message encoding.
+type JSONMessageCodec struct{}
+
+var _ MessageCodec = JSONMessageCodec{}
+
+// EncodeMessage encodes a json message to a slice of bytes.
+func (j JSONMessageCodec) EncodeMessage(message interface{}) (binaryMessage []byte, err error) {
+	return json.Marshal(&message)
```

Comment:
`message interface{}` is already a pointer to something (or a builtin type), Im not sure if we need to take the pointer of message and give it to Marshal(..)

Reply:
Nevermind this.